fe2o3-amqp/src/link/receiver.rs

```diff
@@ -522,9 +519,72 @@ impl Receiver {
         deliveries: impl IntoIterator<Item = impl Into<DeliveryInfo>>,
         modified: Modified,
     ) -> Result<(), DispositionError> {
-        let state = DeliveryState::Modified(modified);
+        let state = TerminalDeliveryState::Modified(modified);
+        self.dispose_all(deliveries, state).await
+    }
+
+    /// Dispose the message by sending a disposition with the provided state
+    ///
+    /// This will not send disposition if the delivery is not found in the local unsettled map.
+    pub async fn dispose(
+        &self,
+        delivery_info: impl Into<DeliveryInfo>,
+        state: TerminalDeliveryState,
```

Comment:
Impl `From` trait to convert `Accepted`, `Rejected`, `Released`, and `Modified` and changing the argument `state: impl Into<TerminalDeliveryState>` might be more ergonomic to use